### ISSUE TITLE
`Struct().from_dict(...)` has no effect

### ISSUE THREAD

**Issue by woodruffw** [danielgtaylor/python-betterproto#549]
### Summary

`protobuf.Struct` appears to not load correctly with `from_dict(...)`, which also affects `from_json(...)`

### Reproduction Steps

I originally saw this with a more complex message, however I've minified it down to a single API call.

Using `betterproto==2.0.0b6`:

```python
>>> import betterproto.lib.google.protobuf as protobuf
>>> protobuf.Struct().from_dict({"a": "b"})
```

### Expected Results

I expect a `Struct` instance populated from the dictionary:

```python
Struct(a="b")
```

### Actual Results

I receive an empty `Struct` instance:

```python
Struct()
```

### System Information

```
libprotoc 25.1
Python 3.12.0
Name: betterproto
Version: 2.0.0b6
Summary: A better Protobuf / gRPC generator & library
Home-page: https://github.com/danielgtaylor/python-betterproto
Author: Daniel G. Taylor
Author-email: danielgtaylor@gmail.com
License: MIT
Location: /Users/william/devel/attestation/python/env/lib/python3.12/site-packages
Requires: grpclib, python-dateutil
Required-by: in-toto-attestation
```

### Checklist

- [X] I have searched the issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have verified this issue occurs on the latest prelease of betterproto which can be installed using `pip install -U --pre betterproto`, if possible.

**Comment by Gobot1234** [comment]
Duplicate of #332

**Comment by woodruffw** [comment]
Correct me if I'm wrong, but I don't believe these are duplicates: the linked issue says that the types aren't supported, while they *do* appear supported but have buggy behavior in the dictionary/JSON conversion cases.

**Comment by woodruffw** [comment]
The follow-up comment on the linked issue is also about a codegen miscompilation, which is not the observed behavior here.

**Comment by Gobot1234** [comment]
We need to have special behaviour for the classes for which there is currently none, this is something else that requires special behaviour, the follow up comment doesn't necessarily mean that it's the same problem as the original issue

**Comment by woodruffw** [comment]
Okay, thanks for clarifying!
